### DISCUSSION TITLE
Expand a virtual machine Pool

### DISCUSSION THREAD

**Discussion by BlackWand1** [minio/minio#18348] [Q&A] [answered]
Hi!

I have 4 machines running in virtual environment. Each machine currently has 4x 10 TB disks (assigned LUN based disks).
Now. I know I could add a completely new pool to the cluster.

But what if I simply extend ALL the 10 TB volymes to a larger size and restart minio?
Increase LUN size and use xfs extend to increase filesystem size. Would minio then recognize the new larger size of all drives in the pool?

Since I believe that adding more number of disks would not be working. Since I would then have a set of disks are are empty while the others have equal usage - because of erasure coding.

**Comment by harshavardhana** [comment, marked as answer]
You can expand the size of the disks and it will be reflected on MinIO.

**Reply by BlackWand1** [reply]
Thanks

**Comment by BlackWand1** [comment]
I am correct in the fact that we cannot add more disks to the pool later on?
The current data will not be "rebalanced"?

So if we buy 4 servers with 24 slots buy only buy 12 drives. Only way to use the 12 remaining slots later on would be a second minio instance with its own erasure coding, correct?

**Reply by klauspost** [reply]
No. It would be a second pool, not a second instance. The cluster remains uniform.

Bigger installs are split into separate sets of drives anyway, so it would effectively close to the same.

**Reply by BlackWand1** [reply]
Yes. But i could run that extra "pool" on the same hardware but another port?
And so use the same hardware, and the new added drives?

**Reply by klauspost** [reply]
You can run it on the same port. 

If you have `server{1...4}:9000/disk{1...4}`, you should be able to add a pool like this: 

`server{1...4}:9000/disk{1...4} server{1...4}:9000/disk{5...8}`.

**Reply by BlackWand1** [reply]
OK. Sounds awesome.
